### PR TITLE
Fix command options

### LINKED ISSUE #239: compareSnapshot command doesn't not consistently handle command options
It seems a major regression was introduced by this change 

https://github.com/cypress-visual-regression/cypress-visual-regression/commit/7b0c9784c6e600ca3cb7f0d81f26ff72e1ff8849#diff-1212ed35aabbfc158720210527b07c5e3ccaf889121fa334acc54b160afe561cL50-L80

Before a function scope var `screenshotAndPluginOptions` was created for each command call where the initial `screenshotOptions` and current command `commandOptions` options would be merged and stored.

```
screenshotAndPluginOptions = { ...screenshotOptions, ...commandOptions }
```

Now the new code is reusing the `addCompareSnapshotCommand` function arg `screenshotOptions` to store global and command `commandOptions`.

```
screenshotOptions = { ...screenshotOptions, ...commandOptions }
```

Which means that every call to `compareSnapshot` would change the original `screenshotOptions` and hence influence the next call of the command, as all options will be carrying on.

So in my tests, I have 

```
  it('should match visual snapshot viewport', () => {
    cy.compareSnapshot('debug-style', {capture: 'viewport'})
  })

  it('should match visual snapshot full', () => {
    cy.compareSnapshot('debug-style')
  })
```

with current code `{capture: 'viewport'}` would be applied to the second test as well.

Generally, this is a common mistake to reuse function arguments for internal manipulation, there is an ESLint rule for it https://eslint.org/docs/latest/rules/no-param-reassign

### PR DESCRIPTION
**Pull request by Zaista**
fixes #239

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,10 +29,10 @@
...
     "clean:snapshots": "rimraf cypress/snapshots",
     "cy:base-and-regression": "npm run cy:base && npm run cy:regression",
-    "cy:open:base": "visual_regression_log=debug cypress open --e2e -b chrome -C cypress.base.config.ts",
-    "cy:open:regression": "visual_regression_log=debug cypress open --e2e -b chrome -C cypress.regression.config.ts",
-    "cy:base": "visual_regression_log=debug cypress run -C cypress.base.config.ts",
-    "cy:regression": "visual_regression_log=debug cypress run -C cypress.regression.config.ts"
+    "cy:open:base": "set visual_regression_log=debug && cypress open --e2e -b chrome -C cypress.base.config.ts",
+    "cy:open:regression": "set visual_regression_log=debug && cypress open --e2e -b chrome -C cypress.regression.config.ts",
+    "cy:base": "set visual_regression_log=debug && cypress run -C cypress.base.config.ts",
+    "cy:regression": "set visual_regression_log=debug && cypress run -C cypress.regression.config.ts"
```
Comment (area73) [inline package.json:35]: This config won't work on Mac.

The good news is that we can use  [cross-env](https://www.npmjs.com/package/cross-env) to solve the problem.

Can you try that approach?   🙏

**Review by area73 — CHANGES_REQUESTED** [review]
I add a comment about package scripts since they are not going to work on a Mac, but maybe we can use  [cross-env](https://www.npmjs.com/package/cross-env) to solve the problem.

**Review by area73 — APPROVED** [review]
👌

**Comment by eugenet8k** [comment]
Thank you @Zaista!

**Pull request merged**